app/controllers/twi_subscriptions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class TwiSubscriptionsController < ApplicationController
+  before_action :redirect_if_not_allowed
+
+  def index
+    # Guard in case someone tries to access the URL without any search results.
+    unless params[:twitter_search].nil?
```

Comment:
So what happens is it is nil? Should it redirect? Show an error? Something else?